### PR TITLE
Feature 6 tree view for html layout

### PR DESCRIPTION
**Pull request by Jdallos**

### CODE REVIEW

File: src/styles/components/layout-screen/styled-tree-item.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+              >
+                { labelText }
+              </Typography>
+              <Typography>
+                { strings.comingSoon }
+              </Typography>
+            </Stack>
+          </div>
+          { isLayoutComponent &&
+            <SvgIcon fontSize="large" sx={{ alignSelf: "center", justifySelf: "center" }}>
```
Comment (Kaemppi) [inline src/styles/components/layout-screen/styled-tree-item.tsx:88]: I'd consider creating an Icon component of this and pass the props.

File: src/components/layout/layout-tree-menu-html.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import { AddBoxOutlined } from '@mui/icons-material';
+import { TreeView } from '@mui/lab';
+import { Stack, Typography } from '@mui/material';
+import StyledTreeItem from '../../styles/components/layout-screen/styled-tree-item';
+import { ComponentType, TreeObject } from '../../types';
```
Comment (EVP1989) [inline src/components/layout/layout-tree-menu-html.tsx:5]:
```suggestion
import { AddBoxOutlined } from "@mui/icons-material";
import { TreeView } from "@mui/lab";
import { Stack, Typography } from "@mui/material";
import StyledTreeItem from "../../styles/components/layout-screen/styled-tree-item";
import { ComponentType, TreeObject } from "../../types";
```

File: src/components/layout/layout-tree-menu-html.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+ * Components properties
+ */
+interface Props {
+  htmlString: string;
+}
+
+/**
+ * Layout Tree Menu HTML Component
+ */
+const LayoutTreeMenuHtml: React.FC<Props> = ({
```
Comment (EVP1989) [inline src/components/layout/layout-tree-menu-html.tsx:17]: Destruct this

File: src/components/layout/layout-tree-menu-html.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        labelText={ item.type }
+        isLayoutComponent={ item.type === ComponentType.LAYOUT }
+        isRoot={ isRoot }
+        hasChildren={ hasChildren }
+      >
+        { item.children.map((x, i) => renderTreeItem(x, undefined)) }
+        { item.type === ComponentType.LAYOUT &&
+            <Stack direction="row" alignItems="center">
+              <AddBoxOutlined/>
+              <Typography variant="caption" textTransform="uppercase">Lisää elementti</Typography>
```
Comment (EVP1989) [inline src/components/layout/layout-tree-menu-html.tsx:75]: Use localized string

File: src/components/layout/layout-tree-menu-html.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    return (
+      <StyledTreeItem
+        key={ item.id }
+        nodeId={ item?.id ?? "" }
+        labelText={ item.type }
+        isLayoutComponent={ item.type === ComponentType.LAYOUT }
+        isRoot={ isRoot }
+        hasChildren={ hasChildren }
+      >
+        { item.children.map((x, i) => renderTreeItem(x, undefined)) }
```
Comment (EVP1989) [inline src/components/layout/layout-tree-menu-html.tsx:71]: Use describin naming, for example  x = child

File: src/components/layout/layout-tree-menu-html.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+            </Stack>
+        }
+      </StyledTreeItem>
+    );
+  };
+
+  return (
+    <TreeView>
+      { Array.isArray(domArray) &&
+        domArray.map((x, i) => {
```
Comment (EVP1989) [inline src/components/layout/layout-tree-menu-html.tsx:85]: Use more describing naming than "x"

File: src/components/screens/layout-screen-html.tsx [hunk 7]
```diff
@@ -19,13 +18,14 @@ import {
 import { WithStyles } from '@mui/styles';
 import withStyles from '@mui/styles/withStyles';
```
Comment (EVP1989) [inline src/components/screens/layout-screen-html.tsx:19]:
```suggestion
import { WithStyles } from "@mui/styles";
import withStyles from "@mui/styles/withStyles";
```

File: src/components/screens/layout-screen-html.tsx [hunk 8]
```diff
@@ -19,13 +18,14 @@ import {
...
 import { KeycloakInstance } from "keycloak-js";
-import { PageLayout, Exhibition, DeviceModel, ScreenOrientation, SubLayout, LayoutType } from "../../generated/client";
+import { PageLayout, Exhibition, DeviceModel, ScreenOrientation, SubLayout, LayoutType, PageLayoutViewHtml } from "../../generated/client";
 import BasicLayout from "../layouts/basic-layout";
 import ElementNavigationPane from "../layouts/element-navigation-pane";
 import EditorView from "../editor/editor-view";
 import { AccessToken, ActionButton, LayoutEditorView } from '../../types';
```
Comment (EVP1989) [inline src/components/screens/layout-screen-html.tsx:25]:
```suggestion
import { AccessToken, ActionButton, LayoutEditorView } from "../../types";
```

**Review by EVP1989 — COMMENTED** [review]
Added comments

File: src/styles/components/layout-screen/styled-tree-item.tsx (hunk 1 above)
Comment (Jdallos) [reply src/styles/components/layout-screen/styled-tree-item.tsx:88]: Created own component but as this is only being used in the one place I thought the props maybe didn't need passing but rather can be contained within the component? Or were you thinking of some of these props in particular that needed passing?

File: src/components/layout/layout-tree-menu-html.tsx [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+    }
+  };
+
+  /**
+   * Renders Tree Item
+   *
+   * @param item item
+   * @param isRoot is root element
+   */
+  const renderTreeItem = (item?: TreeObject, isRoot?: boolean, isRootSubdirectory?: boolean) => {
```
Comment (EVP1989) [inline src/components/layout/layout-tree-menu-html.tsx:60]:
```suggestion
  const renderTreeItem = (item?: TreeObject, isRoot?: boolean, isRootSubDirectory?: boolean) => {
```

File: src/components/layout/layout-tree-menu-html.tsx [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+    const hasChildren = !!item.children.length;
+
+    return (
+      <StyledTreeItem
+        key={ item.id }
+        nodeId={ item?.id ?? "" }
+        labelText={ item.type }
+        isLayoutComponent={ item.type === ComponentType.LAYOUT }
+        isRoot={ isRoot }
+        isRootSubdirectory={ isRootSubdirectory }
```
Comment (EVP1989) [inline src/components/layout/layout-tree-menu-html.tsx:71]:
```suggestion
        isRootSubdirectory={ isRootSubDirectory }
```

File: src/components/layout/layout-tree-menu-html.tsx [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+        hasChildren={ hasChildren }
+      >
+        { item.children.map((child, i) => {
+          const isRootSubdirectory = i === 0;
+          return renderTreeItem(child, false , isRootSubdirectory)
```
Comment (EVP1989) [inline src/components/layout/layout-tree-menu-html.tsx:76]:
```suggestion
          return renderTreeItem(child, false , isRootSubDirectory)
```

**Review by EVP1989 — APPROVED** [review]
Approved with suggestions

**Pull request merged**